internal/commands/server/server.go

```diff
@@ -127,3 +126,23 @@ func RunServer(config ServerConfig) int {
 	config.Logger.Info("shutting down")
 	return 0
 }
+
+func registerSecretClients(config ServerConfig) (*envoy.MultiSecretClient, error) {
+	secretClient := envoy.NewMultiSecretClient()
+
+	k8sSecretClient, err := k8s.NewK8sSecretClient(config.Logger.Named("k8s-cert-fetcher"), config.K8sConfig.RestConfig)
+	if err != nil {
+		config.Logger.Error("error initializing the kubernetes secret fetcher", "error", err)
+		return nil, err
+	}
+	secretClient.Register(utils.K8sSecretScheme, k8sSecretClient)
+
+	vaultSecretClient, err := vault.NewSecretClient(config.Logger.Named("vault-cert-fetcher"), "pki", "TODO")
+	if err != nil {
+		config.Logger.Error("error initializing the vault secret fetcher", "error", err)
+		return nil, err
+	}
+	secretClient.Register(vault.SecretScheme, vaultSecretClient)
```

Comment:
👍 This looks cleaner than the prior `AddToMultiClient` method that mutated the argument passed in.